### ISSUE TITLE
Java API issue on Windows

### ISSUE THREAD

**Issue by DmitryXGIT** [TALP-UPC/FreeLing#24]
FreeLing/APIs/java/readme:
>   HOW TO BUILD THE API IN WINDOWS, USING MSVC 
1. Install java 
2. Download and install swig (http://www.swig.org/)
3. Open msvc project named msvc/10.0/swig/java/freeling_javaAPI

Unfortunately it is impossible to find mentioned msvc project

**Comment by lluisp** [comment]
You are right, that README is outdated.
MSVC project for java API is not provided any more, because it is too dependent on SWIG and Java versions and their installation locations. 

It is not difficult to do it yourself.
See Makefile in APIs/java to get an idea of the needed steps:
  1.- Install SWIG and generate the API
       SWIG will generate a directory edu/upc/freeling with the Java API for all modules, and
       a file freeling_javaAPI.cxx with the JNI side of the API

  2.- Compile java code in edu/upc/freeling and build a jar with the results.
 
  3.- Compile  freeling_javaAPI.cxx  into a DLL (using MSVC or any other compiler).
       You need to provide paths to freeling and treeler include directories, and to Java
       headers (jni.h etc)
       You will get a dll freeling_javaAPI.dll to  be loaded from your Java programs

**Comment by ggiorgetti** [comment]
> 1.- Install SWIG and generate the API
> SWIG will generate a directory edu/upc/freeling with the Java API for all modules, and
> a file freeling_javaAPI.cxx with the JNI side of the API

I run SWIG with the command **swig -java -c++ <_location of freeling_javaAPI.i_>** and it doesn't creates the directory edu/upc/freeling, it just creates a lot of .java files in the same dir that the .i file.
Do I have to use extra parameters for SWIG, or do I have to modify something in the Makefile / freeling_javaAPI.i?
I've been trying to compile this for 2 days now, I really need help, thanks in advance!

**Comment by lluisp** [comment]
I don't think the Makefile will work on windows. So, there is no point in modifying it.

What you need to do is understand what the makefile is doing, and replicate those steps in windows command line. 
If you look at the swig command in the Makefile, you'l see that your command is missing a few options.

 If you have a "make" version for windows, you can try running the makefile, but in any case (either using the makefile or replicating the commands by hand), you  will need to adapt some of the commands (e.g. replace g++ with some other compiler, etc).

  In summary:  You need to understand what the Makefile is doing and adapt it to windows environment and development tools.

**Comment by ggiorgetti** [comment]
Ok, I'll give it one more try. Thanks for your help.
